### DISCUSSION TITLE
Project Reunion loses UWP AOT support - please avoid

### DISCUSSION THREAD

**Discussion by charlesroddie** [dotnet/maui#843] [General]
> Richard Lander: "We will not be enabling .NET Native with Project Reunion.  Any new native AOT model would be based on the NativeAOT experiment"

Current UWP samples use Project Reunion: https://github.com/dotnet/net6-mobile-samples/blob/main/HelloWinUI3/HelloWinUI3/HelloWinUI3.csproj

Please hold off on using Project Reunion until it has support for native AOT. Xamarin.Forms users using UWP need a migration path and migrating to a non-AOT solution isn't acceptable. This is likely to prevent us from migrating to MAUI.

A UWP project based on WinUI is still compatible with .Net Native.

**Comment by ccarlo88** [comment]
There are no plans for having AOT. I am also disappointed but NativeAOT is just experimental and it has no clear roadmap and probably no future. What I can recommend is moving on, there's a nice language very similar to C# which is called Vala and use it with GTK and Glade (for xml UI). It converts the code to C and compile. There's quite a good amount of software made with it.
Source:
https://github.com/dotnet/runtimelab/issues/1120
